src/composeapp.cc

```diff
@@ -42,6 +44,55 @@ void ComposeApp::remove() {
   }
 }
 
+bool ComposeApp::check() {
+  bool result = false;
+
+  do {
+    bool cmd_res{false};
+    std::string cmd_output;
+
+    std::tie(cmd_res, cmd_output) = cmd("config");
+    if (!cmd_res) {
+      LOG_WARNING << "Missing and incomplete App config: " << name_;
+      break;
+    }
+
+    // calculate a number of container images that a given App consists of
+    std::string::size_type find_pos{0};
+    std::string::size_type line_pos{0};
+    int expected_container_number{0};
+
+    while ((find_pos = cmd_output.find("image:", line_pos)) != std::string::npos) {
+      if (cmd_output.substr(line_pos, (find_pos - line_pos)).find_first_of('#') == std::string::npos) {
+        ++expected_container_number;
+      }
+      line_pos = cmd_output.find('\n', find_pos);
+      ++find_pos;
+    }
+
+    // Get a number of running container images
+    std::tie(cmd_res, cmd_output) = cmd("ps -q");
+    if (!cmd_res) {
+      LOG_WARNING << "Failed to get a list of App's containers: " << name_;
+      break;
+    }
+
+    int running_container_number =
+        std::count_if(cmd_output.begin(), cmd_output.end(), [](const char& symbol) { return symbol == '\n'; });
+
+    if (running_container_number < expected_container_number) {
+      LOG_DEBUG << "Number of running containers is less than a number of images specified in the compose file"
+                << "; App: " << name_ << "; expected container number: " << expected_container_number
+                << "; number of running containers: " << running_container_number;
+      break;
+    }
+
+    result = true;
+  } while (false);
```

Comment:
shouldn't this be `while (true)`?

Reply:
no, this statement do/while(false) is used for avoiding nested 'ifs' or multiple returns. I think I will remove this statement at all and just `return false` instead of `break`, this trick is justifiable IMHO if there are really lots of returns and the code is more complex.